test/functional/apps/visualize/_linked_saved_searches.js

```diff
@@ -61,7 +62,9 @@ export default function ({ getService, getPageObjects }) {
         await filterBar.addFilter('bytes', 'is between', '100', '3000');
         await PageObjects.header.waitUntilLoadingHasFinished();
         const data = await PageObjects.visualize.getTableVisData();
-        expect(data.trim()).to.be('707');
+        retry.waitFor('wait for count to change to 707', async () => {
+          expect(data.trim()).to.be('707');
```

Comment:
This isn't going to re-fetch the data on each try, you need to put the call to `getTableVisData()` inside the retry function so it is re-executed. And also, `waitFor()` expects that you return a boolean, like so:

```js
retry.waitFor('wait for count to change to 707', async () => {
  const data = await PageObjects.visualize.getTableVisData();
  return data.trim() === '707'
});
```

If you updated the `getTableVisData()` function to trim the data itself it could be:

```js
retry.waitFor('wait for count to change to 707', async () => (
  await PageObjects.visualize.getTableVisData() === '707'
));
```